src/test/java/uk/gov/hmcts/probate/transformer/SolicitorExecutorTransformerTest.java

```diff
@@ -0,0 +1,191 @@
+                .solsSolicitorIsApplying(CommonVariables.NO)
+                .solsSolicitorNotApplyingReason(CommonVariables.SOLICITOR_NOT_APPLYING_REASON);
+
+        when(caseDetailsMock.getData()).thenReturn(caseDataBuilder.build());
+
+        solicitorExecutorTransformerMock.mainApplicantTransformation(caseDetailsMock.getData(), responseCaseDataBuilder);
+
+        ResponseCaseData responseCaseData = responseCaseDataBuilder.build();
+
+        assertNull(responseCaseData.getSolsSolicitorIsMainApplicant());
+        assertNull(responseCaseData.getSolsSolicitorIsApplying());
+        assertNull(responseCaseData.getSolsSolicitorNotApplyingReason());
+    }
+
+
+    @Test
+    public void shouldRemoveSolicitorDetailsFromPrimaryApplicant(){
+        // Solicitor names are same as primary names.
+        caseDataBuilder
+                .solsSolicitorIsExec(CommonVariables.YES)
+                .solsSolicitorIsMainApplicant(CommonVariables.NO)
+                .solsSolicitorIsApplying(CommonVariables.NO)
+                .solsSOTForenames(CommonVariables.SOLICITOR_SOT_FORENAME)
+                .solsSOTSurname(CommonVariables.SOLICITOR_SOT_SURNAME)
+                .primaryApplicantForenames(CommonVariables.SOLICITOR_SOT_FORENAME)
+                .primaryApplicantSurname(CommonVariables.SOLICITOR_SOT_SURNAME)
+                .primaryApplicantEmailAddress(CommonVariables.SOLICITOR_FIRM_EMAIL)
+                .primaryApplicantPhoneNumber(CommonVariables.SOLICITOR_FIRM_PHONE)
+                .primaryApplicantAddress(CommonVariables.SOLICITOR_ADDRESS)
+                .primaryApplicantAlias(CommonVariables.PRIMARY_EXEC_ALIAS_NAMES)
+                .primaryApplicantHasAlias(CommonVariables.YES)
+                .primaryApplicantIsApplying(CommonVariables.YES)
+                .solsSolicitorNotApplyingReason(CommonVariables.SOLICITOR_NOT_APPLYING_REASON);
+
+        when(caseDetailsMock.getData()).thenReturn(caseDataBuilder.build());
+
+        solicitorExecutorTransformerMock.mainApplicantTransformation(caseDetailsMock.getData(), responseCaseDataBuilder);
+
+        ResponseCaseData responseCaseData = responseCaseDataBuilder.build();
+
+        assertNull(responseCaseData.getPrimaryApplicantForenames());
+        assertNull(responseCaseData.getPrimaryApplicantSurname());
+        assertNull(responseCaseData.getPrimaryApplicantPhoneNumber());
+        assertNull(responseCaseData.getPrimaryApplicantEmailAddress());
+        assertNull(responseCaseData.getPrimaryApplicantAddress());
+        assertNull(responseCaseData.getPrimaryApplicantAlias());
+        assertNull(responseCaseData.getPrimaryApplicantHasAlias());
+        assertNull(responseCaseData.getPrimaryApplicantIsApplying());
+        assertNull(responseCaseData.getSolsPrimaryExecutorNotApplyingReason());
+    }
+
+    @Test
+    public void shouldRemoveSolicitorNotApplyingReasonWhenSolicitorApplying(){
+        caseDataBuilder
+                .solsSolicitorIsExec(CommonVariables.YES)
```

Comment:
you could probably import these common vars to make it more readable

Reply:
Yes good idea